src/io/SPI.cpp

```diff
@@ -1,5 +1,9 @@
 #include <EVT/io/SPI.hpp>
 
+#define SPI_CHECK                \
```

Comment:
I learned about C++ error handling on my co-op, and I think this should look something like the below:
```suggestion
#define SPI_CHECK(functionCall) SPIStatus status = functionCall; if(status != SPIStatus::OK) return status
```
If you want it to be able to be used more than once in  a function, you need to do something like this: https://stackoverflow.com/questions/1082192/how-to-generate-random-variable-names-in-c-using-macros